crates/steel/src/host/db/proof.rs

```diff
@@ -83,6 +89,15 @@ impl<D: Database> ProofDb<D> {
     pub fn inner(&self) -> &D {
         &self.inner
     }
+
+    /// Extends the `ProofDb` with the contents of another `ProofDb`. It panics if they are not
+    /// consistent.
```

Comment:
A `ProofDb` is linked to a single state (commitment) ya? So trying to merge the `ProofDb` for two different blocks will panic here. I can see how, if written correctly, this will never fail. It also seems like an common enough mistake that panicking is perhaps too extreme, and could result in DoS in cases where e.g. someone is running a service that produces Steel proofs, if an error can cause the service to go down.

Reply:
I changed it so that the public extend now returns an error if the header config does not match. This could be a user error.
If the actual hash maps are extended, the panic is kept because this should not be possible without fraudulent RPC nodes. Also, this would be unrecoverable since the DB has already been modified.